src/usecase/tests/category_ucase_test.go

```diff
@@ -66,161 +67,156 @@ func TestAddCategory(t *testing.T) {
-//		})
-//}
+func TestGetCategories(t *testing.T) {
+	r := gofight.New()
+
+	clientRepo := repository.NewClientRepo()
+	userRepo := repository.NewUserRepo()
+	cateringRepo := repository.NewCateringRepo()
+	clientResult, _ := clientRepo.GetByKey("name", "Dymi")
+	userResult, _ := userRepo.GetByKey("email", "admin@meals.com")
+	cateringResult, _ := cateringRepo.GetByKey("name", "Twiist")
+	jwt, _, _ := middleware.Passport().TokenGenerator(&middleware.UserID{userResult.ID.String()})
+	id := cateringResult.ID.String()
+	clientID := clientResult.ID.String()
+	var fakeID = uuid.NewV4()
+
+	// Trying to get categories from non-existing catering
+	// Should throw error
+	r.GET("/caterings/"+fakeID.String()+"/clients/"+clientID+"/categories?date=2020-08-25%2018%3A13%3A55").
+		SetCookie(gofight.H{
+			"jwt": jwt,
+		}).
+		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+			data := []byte(r.Body.String())
+			errorValue, _ := jsonparser.GetString(data, "error")
+			assert.Equal(t, http.StatusNotFound, r.Code)
+			assert.Equal(t, "catering with that ID is not found", errorValue)
+		})
+
+	// Trying to get categories from existing catering
+	// Should be success
+	r.GET("/caterings/"+id+"/clients/"+clientID+"/categories?date=2020-08-25%2018%3A13%3A55").
+		SetCookie(gofight.H{
+			"jwt": jwt,
+		}).
+		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+			assert.Equal(t, http.StatusOK, r.Code)
+		})
+}
+
+func TestUpdateCategory(t *testing.T) {
+	r := gofight.New()
+
+	categoryRepo := repository.NewCategoryRepo()
+	clientRepo := repository.NewClientRepo()
+	userRepo := repository.NewUserRepo()
+	cateringRepo := repository.NewCateringRepo()
+	clientResult, _ := clientRepo.GetByKey("name", "Dymi")
+	userResult, _ := userRepo.GetByKey("email", "admin@meals.com")
+	cateringResult, _ := cateringRepo.GetByKey("name", "Twiist")
+	jwt, _, _ := middleware.Passport().TokenGenerator(&middleware.UserID{userResult.ID.String()})
+	cateringID := cateringResult.ID.String()
+	clientID := clientResult.ID.String()
+
+	// Trying to update non-existing dish category
+	// Should throw an error
+	fakeID := uuid.NewV4()
+	r.PUT("/caterings/"+cateringID+"/clients/"+clientID+"/categories/"+fakeID.String()).
+		SetCookie(gofight.H{
+			"jwt": jwt,
+		}).
+		SetJSON(gofight.D{
+			"name": "zxcvb",
+		}).
+		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+			assert.Equal(t, http.StatusNotFound, r.Code)
+		})
+
+	// Posting new dish category to update it
+	// Should be success
+	r.POST("/caterings/"+cateringID+"/clients/"+clientID+"/categories").
+		SetCookie(gofight.H{
+			"jwt": jwt,
+		}).
+		SetJSON(gofight.D{
+			"name": "qwerty",
+		}).
+		Run(delivery.SetupRouter(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+			assert.Equal(t, http.StatusOK, r.Code)
+		})
+
+	createdCategory, _ := categoryRepo.GetByKey("name", "qwerty", cateringID)
```

Comment:
instead of pulling new category from the database, let's save the result of post method into a variable called categoryID and use it